ouroboros-consensus-cardano-legacy-block/src/Legacy/Cardano/Node.hs

```diff
@@ -0,0 +1,345 @@
+  reconstructHfcNestedCtxt _ sbs sib =
+        conv
+      $ reconstructHfcNestedCtxt
+          (Proxy @(Header (HardForkBlock (CardanoEras c))))
+          sbs
+          sib
+    where
+      conv ::
+           SomeSecond (NestedCtxt Header) (HardForkBlock (CardanoEras c))
+        -> SomeSecond (NestedCtxt Header) (HardForkBlock (LegacyCardanoEras c))
+      conv (SomeSecond ctxt) = SomeSecond $
+          castNestedCtxt hcoerce_NestedCtxt_ ctxt
+
+  getHfcBinaryBlockInfo ::
+       HardForkBlock (LegacyCardanoEras c)
+    -> BinaryBlockInfo
+  getHfcBinaryBlockInfo = getHfcBinaryBlockInfo @(CardanoEras c) . hcoerce_HardForkBlock
+
+  estimateHfcBlockSize ::
+       Header (HardForkBlock (LegacyCardanoEras c))
+    -> SizeInBytes
+  estimateHfcBlockSize = estimateHfcBlockSize @(CardanoEras c) . hcoerce_Header
+
+{-------------------------------------------------------------------------------
+  SupportedNetworkProtocolVersion
+-------------------------------------------------------------------------------}
+
+instance LegacyCardanoHardForkConstraints c
+      => SupportedNetworkProtocolVersion (HardForkBlock (LegacyCardanoEras c)) where
+  supportedNodeToNodeVersions _ = Map.map conv
+      $ supportedNodeToNodeVersions (Proxy @(CardanoBlock c))
+    where
+      conv ::
+           HardForkNodeToNodeVersion (CardanoEras c)
+        -> HardForkNodeToNodeVersion (LegacyCardanoEras c)
+      conv (HardForkNodeToNodeDisabled x) = HardForkNodeToNodeDisabled x
+      conv (HardForkNodeToNodeEnabled x y) =
+          HardForkNodeToNodeEnabled x (htrans (Proxy @ToLegacyBlock) conv' y)
+
+      conv' :: EraNodeToNodeVersion blk -> EraNodeToNodeVersion (LegacyBlock blk)
+      conv' (EraNodeToNodeEnabled x) = EraNodeToNodeEnabled x
+      conv' EraNodeToNodeDisabled    = EraNodeToNodeDisabled
+
+  supportedNodeToClientVersions _ = Map.map conv
+      $ supportedNodeToClientVersions (Proxy @(CardanoBlock c))
+    where
+      conv ::
+           HardForkNodeToClientVersion (CardanoEras c)
+        -> HardForkNodeToClientVersion (LegacyCardanoEras c)
+      conv (HardForkNodeToClientDisabled x) = HardForkNodeToClientDisabled x
+      conv (HardForkNodeToClientEnabled x y) =
+          HardForkNodeToClientEnabled x (htrans (Proxy @ToLegacyBlock) conv' y)
+
+      conv' :: EraNodeToClientVersion blk -> EraNodeToClientVersion (LegacyBlock blk)
+      conv' (EraNodeToClientEnabled x) = EraNodeToClientEnabled x
+      conv' EraNodeToClientDisabled    = EraNodeToClientDisabled
+
+  latestReleasedNodeVersion _ =
+      latestReleasedNodeVersion (Proxy @(CardanoBlock c))
+
+{-------------------------------------------------------------------------------
+  ProtocolInfo
+-------------------------------------------------------------------------------}
+
+-- | Only used in 'protocolInfoLegacyCardano'.
+class (ToLegacyBlock x y, CanStowLedgerTables (LedgerState x)) => C x y
+instance CanStowLedgerTables (LedgerState x) => C x (LegacyBlock x)
```

Comment:
I don't see this being used.

Reply:
It's used here: https://github.com/input-output-hk/ouroboros-consensus/pull/473/files/199d9e308113dc32970549ffe533b967f6e82fe2#diff-165cfbdc83522aa96e78a011a6891f030987b36a63cea4300f1c10ae1e3ed130R253